ssm/lds.py

```diff
@@ -744,26 +904,104 @@ def _fit_laplace_em(self, variational_posterior, datas,
             discrete_expectations = variational_posterior.mean_discrete_states
 
             # 2. Update the continuous state posterior q(x)
+            #    TODO: We are throwing away the log-normalizer from
+            #    variational_posterior.mean_discrete_states() when we coud be
+            #    using it in the ELBO calculations.
             self._fit_laplace_em_continuous_state_update(
                 discrete_expectations, variational_posterior, datas, inputs, masks, tags,
                 continuous_optimizer, continuous_tolerance, continuous_maxiter)
-            continuous_expectations = variational_posterior.sample_continuous_states()
 
-            # 3. Update parameters
+            continuous_samples = variational_posterior.sample_continuous_states()
+            
+            # Update parameters
+            if learning and parameters_update=="exact_mstep":
```

Comment:
how about 
```
if learning:
   if parameters_update == "exact_mstep":
      ...
   elif parameters_update == "stoch_mstep":
      ...
```

Reply:
See my comment below about eliminating branches.  Let's simplify to do exact_mstep where possible and stoch_mstep otherwise.  No more SGD — we can add it back later if we see a compelling use case.